api/envoy/extensions/quic/server_preferred_address/v3/basic_server_preferred_address_config.proto

```diff
@@ -0,0 +1,23 @@
+syntax = "proto3";
+
+package envoy.extensions.quic.server_preferred_address.v3;
+
+import "udpa/annotations/status.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.quic.server_preferred_address.v3";
+option java_outer_classname = "BasicServerPreferredAddressConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/quic/server_preferred_address/v3;server_preferred_addressv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: QUIC server preferred address config]
+// [#extension: envoy.quic.server_preferred_address.basic]
+
+// Configuration for BasicServerPreferredAddressConfig.
+message BasicServerPreferredAddressConfig {
+  // String representation of ipv(4|6) address, i.e. "127.0.0.2".
+  // If not specified, none will be configured.
+  string ipv4_address = 1;
```

Comment:
Since the port of part of the SPA, should this be `message SocketAddress` instead, so that the port can also be configured? Unset/zero could still mean use the listener's port.

Reply:
SocketAddress also has protocol and port fields which are not very meaningful to this extension. And it also just uses a string field to represent ip.

Reply:
Tag @envoyproxy/api-shepherds on this.

Reply:
@envoyproxy/api-shepherds we were discussing this in slack, and wanted to get some more opinions. Right now, due to a single UDP socket (on 0.0.0.0) the port for the main listener and SPA must be the same.

In the future, we may support multiple "linked" sockets for a quic listener for SPA, and then could support different ports for the sockets. What is the best way to allow changing this config in the future to support that? Options we talked about:

1) Create a new extension type for that when it's supported. Leave this config exactly as it currently is.
2) Wrap this config in `oneof` so we can later add `SocketAddress`
3) Wrap this config in `oneof` so we can later add a message with ip string and port int.

Reply:
From a strict forward compatibility perspective:
* Use a `repeated` now if you anticipate wanting multiple later (is this how linked sockets would work), since it's a breaking JSON wire change to switch to repeated later (but not proto3 wire).
* `oneof` can always be added later I believe, it doesn't have any wire implication.

Do you want any of the resolver extension support in `SocketAddress`? FWIW we've found this useful in a number of internal use cases at Google, where the Envoy maps from some name format to the underlying IP. This includes also named port resolution.

Reply:
I don't think we need repeated field here, QUIC only allows configuring 2 SPA, one v4 and one v6.

As to resolver extension support in `SocketAddress`, it hard to tell if we need it or not. The only internal use case we have today will have its own implementation of SPA extension to grab the addresses at runtime instead of statically config them.

Reply:
> * `oneof` can always be added later I believe, it doesn't have any wire implication.

From API_VERSIONING.md, this is disallowed:
```
  * Wrapping an existing field with `oneof`. This has no protobuf or JSON/YAML wire implications,
    but is disruptive to various consuming stubs in languages such as Go, creating unnecessary
    churn.
```

Reply:
How about note this proto with `option (xds.annotations.v3.file_status).work_in_progress = true;` for now? Since we don't have a clear picture of how it is going to be used and what the implementation should be.

Reply:
How shall I move forward on the config API?

Reply:
In fact, I marked the whole extension point of server_preferred_address_config as WIP. We haven't used this feature in production yet and not sure if it will work as expected on the internet. So it's better not to commit to stable API.

Reply:
It's fine if you want to mark it WiP for now, but please add a comment to the effect of "TODO: discuss with API shepherds before removing WiP status".

Reply:
I'd also just "wrap with oneof" for now, @ggreenway is right about the rules.

Reply:
done